appinventor/appengine/src/com/google/appinventor/client/editor/simple/palette/ComponentHelpWidget.java

```diff
@@ -155,6 +162,27 @@ public void setPosition(int offsetWidth, int offsetHeight) {
 
   public ComponentHelpWidget(final SimpleComponentDescriptor scd) {
     super(scd, imageResource);
+    name = scd.getName();
+    helpString = scd.getHelpString();
+    helpURL = scd.getHelpUrl();
+    external = scd.getExternal();
+    version = scd.getVersion();
+    versionName = scd.getVersionName();
+    dateBuilt = scd.getDateBuilt();
+    license = scd.getLicense();
+  }
+
+  public ComponentHelpWidget(String name_p, String helpString_p, String helpURL_p, Boolean external_p,
+                             int version_p, String versionName_p, String datebuilt_p, String license_p) {
+    super(imageResource);
+    name = name_p;
+    helpString = helpString_p;
+    helpURL = helpURL_p;
+    external = external_p;
+    version = version_p;
+    versionName = versionName_p;
+    dateBuilt = datebuilt_p;
+    license = license_p;
```

Comment:
If you really want to have the parameters be named differently from the fields, I would recommend `aName`, `aHelpString`, etc. so that we stick with camel case. Alternatively, you can name them the same and use `this.name = name;`, etc. Of course, if we drop PaletteEntry then this uncalled constructor could go away entirely.

Reply:
> If you really want to have the parameters be named differently from the fields, I would recommend `aName`, `aHelpString`, etc. so that we stick with camel case. Alternatively, you can name them the same and use `this.name = name;`, etc. Of course, if we drop PaletteEntry then this uncalled constructor could go away entirely.

Is there a meaning to the "a" here?